jplag/src/main/java/de/jplag/options/JPlagOptions.java

```diff
@@ -1,320 +1,265 @@
     }
 
     public boolean hasBaseCode() {
-        return this.baseCodeSubmissionName != null;
-    }
-
-    public boolean isDebugParser() {
-        return debugParser;
-    }
-
-    public ClusteringOptions getClusteringOptions() {
-        return this.clusteringOptions;
-    }
-
-    public void setBaseCodeSubmissionName(String baseCodeSubmissionName) {
-        if (baseCodeSubmissionName == null || baseCodeSubmissionName.isEmpty()) {
-            this.baseCodeSubmissionName = null;
-        } else {
-            this.baseCodeSubmissionName = baseCodeSubmissionName;
-        }
+        return baseCodeSubmissionName != null && !baseCodeSubmissionName.isBlank();
     }
 
-    public void setComparisonMode(ComparisonMode comparisonMode) {
-        this.comparisonMode = comparisonMode;
-    }
-
-    public void setDebugParser(boolean debugParser) {
-        this.debugParser = debugParser;
+    public Set<String> excludedFiles() {
+        return Optional.ofNullable(exclusionFileName()).map(this::readExclusionFile).orElse(Collections.emptySet());
     }
 
-    public void setExcludedFiles(Set<String> excludedFiles) {
-        this.excludedFiles = excludedFiles;
-    }
-
-    public void setExclusionFileName(String exclusionFileName) {
-        this.exclusionFileName = exclusionFileName;
-    }
-
-    public void setFileSuffixes(String[] fileSuffixes) {
-        this.fileSuffixes = fileSuffixes;
-    }
-
-    public void setLanguage(Language language) {
-        this.language = language;
+    @Override
+    public List<String> fileSuffixes() {
+        var language = language();
+        if (fileSuffixes == null && language != null)
+            return Arrays.stream(language.suffixes()).toList();
+        return fileSuffixes == null ? null : fileSuffixes.stream().toList();
     }
 
     /**
-     * After the selected language has been initialized, this method is called by JPlag to set default values for options
-     * not set by the user.
-     * @param language - initialized language instance
+     * Path name of the directory containing the base code.<br>
+     * For backwards compatibility it may also be a directory name inside the root directory. Condition for the latter is
+     * <ul>
+     * <li>Specified path does not exist.</li>
+     * <li>Name has not have a separator character after trimming them from both ends (leaving at least a one-character
+     * name).</li>
+     * <li>A submission with the specified name exists in the root directory.</li>
+     * </ul>
+     * It's an error if a string has been provided, but it is neither an existing path nor does it fulfill all the
+     * conditions of the compatibility fallback listed above.
      */
-    public void setLanguageDefaults(Language language) {
-        if (!hasMinimumTokenMatch()) {
-            setMinimumTokenMatch(language.minimumTokenMatch());
+    @Override
+    public String baseCodeSubmissionName() {
+        return baseCodeSubmissionName;
+    }
+
```

Comment:
```suggestion
```

Reply:
This is just for the documentation of the getter. I don't have a better idea here

Reply:
Documented in https://github.com/jplag/JPlag/issues/625